woocommerce/woocommerce-gutenberg-products-block
# load all JSON translation files correctly

This is a superset https://github.com/woocommerce/woocommerce-blocks/pull/7995

While it's hard to fully reproduce this PR (because it touches translation side we might know where) it does indeed prevent a problem of translation not loading correctly, mainly around the express payment block.

As to why this works:
When webpack produces chunks, it sometimes produces shared files between several files, those files sometimes live in a different folder (shared chunk between Cart and Checkout but lives in Cart folder). In this case, Checkout will fail to load its translation. This PR fixes that.

It builds on Niels fix in which we were loading translation from an old place, kinda related.
### Testing steps

1. Switch to Dutch in your website.
2. Make sure all translation is working fine (in Shipping, summary items, and express payment).
